python/cudf/cudf/_lib/orc.pyx

```diff
@@ -75,10 +217,29 @@ cpdef read_raw_orc_statistics(filepath_or_buffer):
     if isinstance(filepath_or_buffer, NativeFile):
         filepath_or_buffer = NativeFileDatasource(filepath_or_buffer)
 
-    cdef raw_orc_statistics raw = (
-        libcudf_read_raw_orc_statistics(make_source_info([filepath_or_buffer]))
+    cdef parsed_orc_statistics parsed = (
+        libcudf_read_parsed_orc_statistics(make_source_info([filepath_or_buffer]))
     )
-    return (raw.column_names, raw.file_stats, raw.stripes_stats)
+
+    cdef vector[column_statistics] file_stats = parsed.file_stats
+    cdef vector[vector[column_statistics]] stripes_stats = parsed.stripes_stats
+    parsed_file_stats = []
+    parsed_stripes_stats = []
+
+    for column_index in range(file_stats.size()):
+        parsed_file_stats.append(
+            _parse_column_type_statistics(file_stats[column_index])
+        )
```

Comment:
List comp?

Reply:
Yes, this is mirroring the old logic that did more work. I'll change to list comprehensions here and below.

Reply:
I do like using list comprehensions here but I have to use indexing like `[f(v[i]) for i in v.size()]` rather than `[f(x) for x in v]` because the latter form requires a nullary default constructor for `column_statistics`, which I don't really want to add.

See commit 86776571c5a3a63b939cdcd6001b436d80e4cd2f.

Reply:
I agree that integer indexing is the right solution here. We shouldn't add a nullary constructor just to make Cython happy (although the reality is that we've definitely done this in the past, see https://github.com/search?q=repo%3Arapidsai%2Fcudf+cython+default+path%3Acpp&type=code).